File: app/src/main/java/com/example/financialtransactions/db/PlaidItemDao.java
package com.example.financialtransactions.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import com.example.financialtransactions.model.PlaidItem;
import java.util.List;

@Dao
public interface PlaidItemDao {
    @Query("SELECT * FROM plaid_items")
    List<PlaidItem> getAllItems();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertItem(PlaidItem item);

    @Query("DELETE FROM plaid_items WHERE itemId = :itemId")
    void deleteByItemId(String itemId);
}

File: app/src/main/java/com/example/financialtransactions/db/PlaidItemDao.kt
package com.example.financialtransactions.db

import androidx.room.*
import com.example.financialtransactions.model.PlaidItem

@Dao
interface PlaidItemDao {
    @Query("SELECT * FROM plaid_items")
    suspend fun getAllItems(): List<PlaidItem>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertItem(item: PlaidItem)

    @Delete
    suspend fun deleteItem(item: PlaidItem)

    @Query("DELETE FROM plaid_items WHERE itemId = :itemId")
    suspend fun deleteByItemId(itemId: String)
}
